ghostratings/calculations.php

```diff
@@ -0,0 +1,346 @@
+<?php
+/*
+    Copyright (C) 2004-2010 Kestas J. Kuliukas
+
+	This file is part of webDiplomacy.
+
+    webDiplomacy is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    webDiplomacy is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with webDiplomacy.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+defined('IN_CODE') or die('This script can not be run by itself.');
+
+/**
+ * Code that calculates and alters Ghost Ratings
+ *
+ */
+ class GhostRatings
+ {  
+   private $gameID;
+   private $SCcounts;
+   private $memberStatus;
+   private $variantID;
+   private $pressType;
+   private $potType;
+   private $gameTurns;
+   private $gameStatus;
+   private $phaseMinutes;
+   private $victorySC;
+   private $variantSC;
+   private $winner;
+   private $time;
+   
+   private $variantMod;
+   private $pressMod;
+   private $modValue;
+   
+   private $k;
+   private $start;
+   private $modMultiplier;
+
+ 	
+ 	public function __construct($gameID, $SCcounts, $memberStatus, $variantID, $pressType, $potType, $gameTurns, $gameStatus, $phaseMinutes, $victorySC, $variantSC, $winner, $botGame, $time)
+   {
+     $this->gameID = $gameID;
+     $this->SCcounts = $SCcounts;
+     $this->memberStatus = $memberStatus;
+     $this->variantID = $variantID;
+     $this->pressType = $pressType;
+     $this->potType = $potType;
+     $this->gameTurns = $gameTurns;
+     $this->gameStatus = $gameStatus;
+     $this->phaseMinutes = $phaseMinutes;
+     $this->variantMod = Config::$grVariantMods[$this->variantID];
+     $this->victorySC = $victorySC;
+     $this->variantSC = $variantSC;
+     $this->winner = $winner;
+     $this->botGame = $botGame;
+     $this->time = $time;
+     $this->k = 32;
+     $this->start = 100;
+     $this->pressMod = Config::$grPressMods[$this->pressType];
+     $this->modvalue = (17.5 * $this->variantMod * $this->pressMod);
```

Comment:
What is 17.5? Can you throw in some comments around any of the number variables here llike 32, 100, etc. Will help anyone else like vDip who takes this to adjust a rating to their site needs.

Reply:
All of these variables are just things used to control the scaling of the ranking systems: where a players rating starts at, how much they move up and down per game, etc. I will add comments on my next commit specifying this in the file.

Reply:
Just put in a commit which addresses this.